jitbuilder/control/Jit.cpp

```diff
@@ -122,11 +124,19 @@ initializeJitBuilder(TR_RuntimeHelper *helperIDs, void **helperAddresses, int32_
    //
    TR::RawAllocator rawAllocator;
 
+   // initialize and attach current thread to thread library, and then initialize port library
+   if (0 == omrthread_init_library())
+      {
+      omrthread_t currentThread = NULL;
+      if (0 == omrthread_attach_ex(&currentThread, J9THREAD_ATTR_DEFAULT))
+         omrport_init_library(&portLibrary, sizeof(OMRPortLibrary));
+      }
+
```

Comment:
I seem to recall a debate in the early days of JB and portlib dependence as to whether JB should allocate its own port library or accept it as an initialization parameter, but I don't recall if there was a clear opinion that emerged from that. If a port library is now required for functionality (the build dependence was already set up years ago), then the answer might be "both" (allocate one if it isn't provided as a parameter).

We've talked about doing work to reduce the surface area of the port library (by re-arranging the functions in the function table to only the bare minimum required for OMR, for example). Including a port library in JB would seem to raise the priority of that work.

Tagging @mstoodle in case he has an opinion on this in the current JB and the future JB2.

Reply:
This PR makes the port library a requirement.  I think `initializeJitBuilder()` can be amended to take an optional port library pointer and use it if one is provided, but if it is null then JB will allocate one (and free it) as Brad has done in this PR.